src/Raven.Server/Web/Studio/SqlMigrationHandler.cs

```diff
@@ -0,0 +1,173 @@
+                }
+            }
+            return Task.CompletedTask;
+        }
+
+
+        [RavenAction("/databases/*/admin/sql-migration", "POST", AuthorizationStatus.DatabaseAdmin)]
+        public async Task ImportSql()
+        {
+            using (ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
+            {
+                var sqlImportDoc = context.ReadForDisk(RequestBodyStream(), null);
+
+                if (sqlImportDoc.TryGet("ConnectionString", out string connectionString) == false)
+                    throw new InvalidOperationException("'ConnectionString' is a required field when asking for sql-migration");
+
+                if (!ValidateConnection(connectionString))
+                {
+                    WriteRespone(new List<string> { "Cannot open connection using the given connection string" }, context);
+                    return;
+                }
+
+                if (sqlImportDoc.TryGet("Tables", out BlittableJsonReaderArray tablesFromUser) == false)
+                    throw new InvalidOperationException("'Tables' is a required field when asking for sql-migration");
+
+                var options = new RavenDocumentFactory.WriteOptions();
+
+                if (sqlImportDoc.TryGet("BinaryToAttachment", out bool binaryToAttachment))
+                    options.BinaryToAttachment = binaryToAttachment;
+
+                if (sqlImportDoc.TryGet("IncludeSchema", out bool includeSchema))
+                    options.IncludeSchema = includeSchema;
+
+                if (sqlImportDoc.TryGet("TrimStrings", out bool trimStrings))
+                    options.TrimStrings = trimStrings;
+
+                if (sqlImportDoc.TryGet("SkipUnsupportedTypes", out bool skipUnsopportedTypes))
+                    options.SkipUnsopportedTypes = skipUnsopportedTypes;
+
+                var factory = new RavenDocumentFactory(options);
+
+                var database = new SqlDatabase(connectionString, Database, factory, tablesFromUser);
+
+                database.Validate(out var errors);
+
+                if (database.IsValid())
+                {
+                    using (var writer = new RavenWriter(context, database))
+                    {
+                        writer.OnTableWritten += OnTableWritten;
+
+                        try
+                        {
+                            await writer.WriteDatabase();
+                        }
+                        catch (Exception e)
+                        {
+                            errors.Add(e.Message);
+                        }
+                    }
+                }
+
+                WriteRespone(errors, context);
+            }
+        }
+
+        private bool ValidateConnection(string connectionString)
+        {
+            SqlConnection connection;
+
+            try
+            {
+                connection = (SqlConnection) ConnectionFactory.OpenConnection(connectionString);
+            }
+            catch
```

Comment:
This is horrible. You know what the error is, you won't tell me, and you lose information along the way.
Users hate this kind of behavior. We need to always be aware of our error conditions and our error _reporting_.